Description: set CMAKE_LIBRARY_ARCHITECTURE and switch to multiarch support.
Author: Nobuhiro Iwamatsu <iwamatsu@debian.org>
Last-Update: 2012-08-08
Forwarded: no

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -131,7 +131,7 @@ endif()
 if(ANDROID)
     set(OPENCV_LIB_INSTALL_PATH libs/${ARMEABI_NDK_NAME})
 else()
-    set(OPENCV_LIB_INSTALL_PATH lib${LIB_SUFFIX})
+    set(OPENCV_LIB_INSTALL_PATH lib/${CMAKE_LIBRARY_ARCHITECTURE})
 endif()
 
 set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${OPENCV_LIB_INSTALL_PATH}")
--- a/OpenCVModule.cmake
+++ b/OpenCVModule.cmake
@@ -111,6 +111,11 @@ macro(define_opencv_module name)
         RUNTIME DESTINATION bin COMPONENT main
         LIBRARY DESTINATION ${OPENCV_LIB_INSTALL_PATH} COMPONENT main
         ARCHIVE DESTINATION ${OPENCV_LIB_INSTALL_PATH} COMPONENT main)
+ 
+	install(TARGETS ${the_target_s}
+        RUNTIME DESTINATION bin COMPONENT main
+        LIBRARY DESTINATION ${OPENCV_LIB_INSTALL_PATH} COMPONENT main
+        ARCHIVE DESTINATION ${OPENCV_LIB_INSTALL_PATH} COMPONENT main)
 
     install(FILES ${lib_hdrs}
         DESTINATION ${OPENCV_INCLUDE_PREFIX}/opencv2/${name}
